### ISSUE TITLE
[Bug] Can't install customized apps

### ISSUE THREAD

**Issue by georgerappel** [IceWhaleTech/CasaOS#1737]
**Describe the bug**

Can't install a customized app. Both pasting and uploading a docker-compose file have weird behavior.

**To Reproduce**

> Steps to reproduce the behavior:

1. Go to the homepage
2. Click on 'the "+" sign on the top right side of the app list
3. Click on "Install a customized app"
4. Click on the first icon on the top right, "Import".
5. Paste a docker-compose.yaml content ([this](https://github.com/WisdomSky/CasaOS-LinuxServer-AppStore/blob/main/Apps/Babybuddy/docker-compose.yml) was the last random one I tried)
6. Click "Submit"
7. A pop-up appears (screenshot below)

**Expected behavior**

> If I remember correctly, it should:
7. Go back to the full app form
8. Have most of the information filled

**Screenshots**

<img width="802" alt="image" src="https://github.com/IceWhaleTech/CasaOS/assets/7808464/d6f96173-dfcc-4602-9c48-776e026b72b1">

**Desktop (please complete the following information):**

```
 - OS: MacOS M1
 - Browser Chrome
 - Version 122.0.6261.112 (Official Build) (arm64)
```

**System Time**

```
Local time: Fri 2024-03-15 01:21:05 UTC
           Universal time: Fri 2024-03-15 01:21:05 UTC
                 RTC time: Fri 2024-03-15 01:21:05    
                Time zone: Etc/UTC (UTC, +0000)       
System clock synchronized: yes                        
              NTP service: active                     
          RTC in local TZ: no 
```

**Logs**

> Run following command to collect corresponding logs:

```bash
$ sudo journalctl -xef -u casaos-gateway
Mar 15 00:45:29 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 00:45:29 | 200 |   14.900486ms |     192.168.0.2 | GET      "/fonts/materialdesignicons-webfont.66e43f0f.woff2"
Mar 15 00:47:52 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 00:47:52 | 200 |    5.685389ms |     192.168.0.2 | GET      "/img/default.ee0295e7.svg"
Mar 15 01:12:43 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 01:12:43 | 304 |     276.535µs |     192.168.0.2 | GET      "/img/success.066bb944.svg"
Mar 15 01:12:46 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 01:12:46 | 200 |    4.350844ms |     192.168.0.2 | GET      "/"
Mar 15 01:12:46 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 01:12:46 | 304 |     240.953µs |     192.168.0.2 | GET      "/favicon.ico"
Mar 15 01:12:47 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 01:12:47 | 200 |    1.704814ms |     192.168.0.2 | GET      "/v1/gateway/port"
Mar 15 01:14:57 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 01:14:57 | 200 |    3.765126ms |     192.168.0.2 | GET      "/"
Mar 15 01:14:58 casacopa casaos-gateway[996]: [GIN] 2024/03/15 - 01:14:58 | 200 |     8.93609ms |     192.168.0.2 | GET      "/v1/gateway/port"

$ sudo journalctl -xef -u casaos-user-service
Mar 15 01:14:58 casacopa casaos-user-service[1290]: [GIN] 2024/03/15 - 01:14:58 | 200 |    1.814183ms |     192.168.0.2 | GET      "/v1/users/current/custom/widgets_config"
Mar 15 01:14:58 casacopa casaos-user-service[1290]: [GIN] 2024/03/15 - 01:14:58 | 200 |     412.243µs |     192.168.0.2 | GET      "/v1/users/current/custom/link"
Mar 15 01:14:58 casacopa casaos-user-service[1290]: [GIN] 2024/03/15 - 01:14:58 | 200 |     428.623µs |     192.168.0.2 | GET      "/v1/users/current/custom/app_order"

$ sudo journalctl -xef -u casaos-local-storage
ar 14 10:18:32 casacopa casaos-local-storage[1338]: 2024-03-14T10:18:32.108Z        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
Mar 15 00:39:31 casacopa casaos-local-storage[1338]: 2024-03-15T00:39:31.043Z        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
Mar 15 01:12:47 casacopa casaos-local-storage[1338]: 2024-03-15T01:12:47.167Z        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
Mar 15 01:14:58 casacopa casaos-local-storage[1338]: 2024-03-15T01:14:58.650Z        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}

$ sudo journalctl -xef -u casaos-app-management


$ sudo journalctl -xef -u casaos.service
Mar 15 01:13:55 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:00 casacopa casaos[1260]: 消息来了，message：{"type":"ping"}
Mar 15 01:14:00 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:00 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:05 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:05 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:10 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:10 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:15 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:15 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:20 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:20 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:25 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:25 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:30 casacopa casaos[1260]: 消息来了，message：{"type":"ping"}
Mar 15 01:14:30 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:30 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:35 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:35 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:40 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:40 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:45 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:45 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:50 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:50 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:55 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:55 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:58 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:14:58 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:14:58 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;GetDeviceTree
Mar 15 01:14:58 casacopa casaos[1260]: exit status 1
Mar 15 01:15:00 casacopa casaos[1260]: 消息来了，message：{"type":"ping"}
Mar 15 01:15:00 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:15:00 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:15:05 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
Mar 15 01:15:05 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState enp3s0
Mar 15 01:15:10 casacopa casaos[1260]: /bin/bash -c source /usr/share/casaos/shell/helper.sh ;CatNetCardState wlp2s0
```

**Comment by plangin** [comment]
I checked your provided Babybuddy file and found the bug:

When you import the file, the online form seems to require the published port in brackets. So if you fix it yourself into

```
    ports:
      - target: 8000
        published: **"8000"**
```

... it will work.

But I do agree here, the importer should show some error message and the line where the error occurs instead of doing just nothing.

**Comment by georgerappel** [comment]
Thanks @plangin, It did work with the quotes! But importing from the App Store directly works normally from those same files through the custom app store. When clicking install, it fills the info correctly on that page for the install and it works. Not sure if it's an issue with the parser when pasting, or if it's expected anyway.

**Comment by georgerappel** [comment]
Hey guys, It seems like it was fixed on version 0.4.8, right?

 thanks again for the support!
